### PR TITLE
Pipe following demo

### PR DESCRIPTION
**Pull request by ignaciotb**

### CODE REVIEW

**Comment by nilsbore** [comment]
Ok, you need to resolve the conflicts before merging. Otherwise looking good. Starting to use ceres, eh?

**Comment by ignaciotb** [comment]
Merged conflicts, take a look before I merge

File: auv_ekf_slam/src/ekf_slam_core.cpp [hunk 1]
```diff
@@ -352,12 +350,13 @@ void EKFCore::sequentialUpdate(CorrespondenceClass const& c_i_j, Eigen::MatrixXd
...
     Eigen::MatrixXd aux_mat = (Eigen::MatrixXd::Identity(temp_sigma.rows(), temp_sigma.cols()) - K_t_i * c_i_j.H_t_) * temp_sigma;
     Sigma_hat_.block(0,0,6,6) = aux_mat.block(0,0,6,6);
+
     Sigma_hat_.block((c_i_j.i_j_.second) * 3 + 6, (c_i_j.i_j_.second) * 3 + 6, 3, 3) = aux_mat.block(6, 6, 3, 3);
     Sigma_hat_.block((c_i_j.i_j_.second) * 3 + 6, 0, 3, 6) = aux_mat.block(6,0,3,6);
     Sigma_hat_.block(0, (c_i_j.i_j_.second) * 3 + 6, 6, 3) = aux_mat.block(0,6,6,3);
 }
 
-std::tuple<Eigen::VectorXd, Eigen::MatrixXd> EKFCore::ekfUpdate(){
+std::tuple<Eigen::VectorXd, Eigen::MatrixXd, std::vector<Eigen::Vector3d> > EKFCore::ekfUpdate(){
```
Comment (nilsbore) [inline auv_ekf_slam/src/ekf_slam_core.cpp:359]: Don't you get the same problem with segfaults with this as you got the other day?

File: auv_ekf_slam/include/ekf_slam_core/ekf_slam_core.hpp [hunk 2]
```diff
@@ -64,6 +65,7 @@ class EKFCore{
     double mh_dist_fls_;
     double mh_dist_mbes_;
     int lm_num_;
+    std::vector<Eigen::Vector3d> map_pipe_;
```
Comment (nilsbore) [inline auv_ekf_slam/include/ekf_slam_core/ekf_slam_core.hpp:68]: Same question here

File: auv_ekf_slam/include/ekf_slam_core/ekf_slam_core.hpp [hunk 3]
```diff
@@ -38,8 +38,9 @@ class EKFCore{
     EKFCore(Eigen::VectorXd& mu, Eigen::MatrixXd& Sigma, Eigen::MatrixXd& R, Eigen::MatrixXd& Q_fls, Eigen::MatrixXd &Q_mbes,
             double& lambda_fls, double& lambda_mbes, tf::StampedTransform& tf_base_sensor, const double mh_dist_fls, const double mh_dist_mbes);
     ~EKFCore();
-    std::tuple<Eigen::VectorXd, Eigen::MatrixXd> ekfUpdate();
+    std::tuple<Eigen::VectorXd, Eigen::MatrixXd, std::vector<Eigen::Vector3d>> ekfUpdate();
```
Comment (nilsbore) [inline auv_ekf_slam/include/ekf_slam_core/ekf_slam_core.hpp:41]: Same thing here.

**Review by nilsbore — COMMENTED** [review]
Ok, I can't really judge but it's looking OK, except for the allocator thingy.

File: auv_ekf_slam/src/ekf_slam_core.cpp (hunk 1 above)
Comment (ignaciotb) [reply auv_ekf_slam/src/ekf_slam_core.cpp:359]: No, seems like the other day I needed to add the macro EIGEN_MAKE_ALIGNED_OPERATOR_NEW to the struct I was storing in vectors for this to work. This works fine

**Pull request merged**